### PR TITLE
feat: Distinguish Objects from Veli.

### PR DESCRIPTION
**Pull request by n1k0**
## :wrench: Problem

Ecobalyse don't feature a simulator for Veli ([Notion Card](https://www.notion.so/VeLI_-S-lection-de-la-cat-gorie-VAE-v0-phase-de-production-ac7d63640c7444b9a314667ef7dcb001))

## :cake: Solution

Add a Veli menu entry and add scope information to Veli examples to help distinguish them from other Objects. 

## :desert_island: How to test

Check that both menu entries point to expected pages and features (including bookmarks, url/permalinks and so on).

### CODE REVIEW

**Comment by n1k0** [comment]
@vjousse so after validating with POs, we actually want to separate Objects and Veli pages and features while using a few transverse components, which is what this patch implements. It also introduces `ENABLE_OBJECT_SECTION` and `ENABLE_VELI_SECTION` env vars so we can define which sections should be enabled and served on a given instance.

File: index.js [hunk 1]
```diff
@@ -37,8 +37,12 @@ const storeKey = "store";
 const app = Elm.Main.init({
   flags: {
     clientUrl: location.origin + location.pathname,
-    enableFoodSection: process.env.ENABLE_FOOD_SECTION === "True",
-    enableObjectSection: process.env.ENABLE_OBJECT_SECTION === "True",
+    enabledSections: {
+      food: process.env.ENABLE_FOOD_SECTION === "True",
+      objects: process.env.ENABLE_OBJECT_SECTION === "True",
```
Comment (vjousse) [inline index.js:42]: Maybe we should use `object` or `objects` everywhere but not mixing and matching both?

File: src/Data/Bookmark.elm [hunk 2]
```diff
@@ -30,21 +32,33 @@ type alias Bookmark =
     { created : Posix
     , name : String
     , query : Query
+    , subScope : Maybe Scope
```
Comment (vjousse) [inline src/Data/Bookmark.elm:35]: Why not `scope` directly?

File: src/Data/Scope.elm [hunk 3]
```diff
@@ -70,6 +74,9 @@ toLabel scope =
         Textile ->
             "Textile"
 
+        Veli ->
+            "Véhicule intermédiaire"
```
Comment (vjousse) [inline src/Data/Scope.elm:78]: What about "Véhicules intermédiaires" to be consistent with "Objets"?

File: src/Route.elm [hunk 4]
```diff
@@ -278,3 +294,12 @@ toString route =
                     [ "textile", "simulator" ]
     in
     "#/" ++ String.join "/" pieces
+
+
+objectPrefixPath : Scope -> String
```
Comment (vjousse) [inline src/Route.elm:299]: Maybe use the `toString` method of `Scope` here?

File: src/Views/Page.elm [hunk 5]
```diff
@@ -133,20 +133,25 @@ newVersionAlert { session, reloadPage } =
...
-          else
-            Nothing
+        , addRouteIf enabledSections.textile <|
+            Internal "Textile" Route.TextileSimulatorHome TextileSimulator
+        , addRouteIf enabledSections.food <|
+            Internal "Alimentaire" Route.FoodBuilderHome FoodBuilder
+        , addRouteIf enabledSections.objects <|
+            Internal "Objets" (Route.ObjectSimulatorHome Scope.Object) (Object Scope.Object)
+        , addRouteIf enabledSections.veli <|
+            Internal "Véhicule intermédiaire" (Route.ObjectSimulatorHome Scope.Veli) (Object Scope.Veli)
```
Comment (vjousse) [inline src/Views/Page.elm:154]: Same here, maybe use the plural form?

File: tests/Views/FormatTest.elm [hunk 6]
```diff
@@ -19,8 +19,8 @@ suite =
                 |> asTest "should format zero"
             , 5
                 |> Format.formatFloat 2
-                |> Expect.equal "5,00"
-                |> asTest "should not format an int rendering a specific number of 0 decimals"
```
Comment (vjousse) [inline tests/Views/FormatTest.elm:23]: Why did you remove this test?

**Review by vjousse — APPROVED** [review]
LGTM, just some nits :+1:

File: src/Data/Bookmark.elm (hunk 2 above)
Comment (n1k0) [reply src/Data/Bookmark.elm:35]: Because it helps disambiguating the scope of the page and the scope of the bookmark, which are different things. A page scoped with "objects" might list veli bookmarks.

File: src/Views/Page.elm (hunk 5 above)
Comment (n1k0) [reply src/Views/Page.elm:154]: Ah, now I get it.

File: src/Data/Scope.elm (hunk 3 above)
Comment (n1k0) [reply src/Data/Scope.elm]: Got it.
